src/__tests__/pick.test.ts

```diff
@@ -131,4 +131,32 @@ describe('pick', () => {
         const errors = validateSync(obj);
         expect(errors).toHaveLength(0);
     });
+    it('redefines a field from a grand parent', () => {
```

Comment:
This test case is based on Ran Halprin's failing example [here](https://github.com/hippo-oss/nest-dto/compare/main...ranhalprin:tmp/repro_pick_excludes_extended?expand=1). I made the test a bit more complex by changing the type of the field.